R/use_crate.R

```diff
@@ -43,14 +45,15 @@ use_crate <- function(
     git = NULL,
     version = NULL,
     optional = FALSE,
-    path = ".") {
-  # check args
-  check_string(crate)
-  check_character(features, allow_null = TRUE)
-  check_string(git, allow_null = TRUE)
-  check_string(version, allow_null = TRUE)
-  check_bool(optional)
-  check_string(path)
+    path = ".",
+    echo = TRUE) {
+  check_string(crate, class = "rextendr_error")
+  check_character(features, allow_null = TRUE, class = "rextendr_error")
+  check_string(git, allow_null = TRUE, class = "rextendr_error")
+  check_string(version, allow_null = TRUE, class = "rextendr_error")
+  check_bool(optional, class = "rextendr_error")
+  check_string(path, class = "rextendr_error")
+  check_bool(echo, class = "rextendr_error")
```

Comment:
Do we anywhere use different `class`? Why not embed it into functions by default? Or completely?

Reply:
We don't. I think we can address this in a followup PR? One of the  challenges i foresee is that class is set via `...` here. The bulk of these checks have moved into the `run_cargo()` which reduces amount of the `class = "rextendr_error"` instances